attestationreport/tdx.go

```diff
@@ -0,0 +1,474 @@
+	if err != nil {
+		return fmt.Errorf("failed to parse QE Report: %v", err)
+	}
+
+	err = binary.Read(buf, binary.LittleEndian, &sig.QECertData.QEReportSignature)
+	if err != nil {
+		return fmt.Errorf("failed to parse QE Report Signature: %v", err)
+	}
+
+	err = binary.Read(buf, binary.LittleEndian, &sig.QECertData.QEAuthDataSize)
+	if err != nil {
+		return fmt.Errorf("failed to parse QE Authentication Data Size: %v", err)
+	}
+
+	tmp := make([]byte, sig.QECertData.QEAuthDataSize)
+	err = binary.Read(buf, binary.LittleEndian, &tmp)
+	if err != nil {
+		return fmt.Errorf("failed to parse QE Authentication Data: %v", err)
+	}
+	sig.QECertData.QEAuthData = tmp
+
+	// parse QE Certification Data Type (has to be type 5)
+	err = binary.Read(buf, binary.LittleEndian, &sig.QECertData.QECertDataType)
+	if err != nil {
+		return fmt.Errorf("failed to parse QE Certification Data Type: %v", err)
+	}
+	if sig.QECertData.QECertDataType != 5 {
+		return fmt.Errorf("wrong QECertDataType. Expected: 5, Got: %v", sig.QECertData.QECertDataType)
+	}
+
+	err = binary.Read(buf, binary.LittleEndian, &sig.QECertData.QECertDataSize)
+	if err != nil {
+		return fmt.Errorf("failed to parse QE Certification Data Size: %v", err)
+	}
+
+	// parse QE Certification Data (Raw)
+	tmp = make([]byte, sig.QECertData.QECertDataSize)
+	err = binary.Read(buf, binary.LittleEndian, &tmp)
+	if err != nil {
+		return fmt.Errorf("failed to parse QE Certification Data: %v", err)
+	}
+
+	// parse PCK Cert Chain (PCK Leaf Cert || Intermediate CA Cert || Root CA Cert)
+	certChain, err := parseCertificates(tmp[:], true)
+	if err != nil {
+		return fmt.Errorf("failed to parse certificate chain from QECertData: %v", err)
+	}
+	sig.QECertData.QECertData = certChain
+
+	return nil
+}
+
+func verifyTdxMeasurements(tdxM *TdxMeasurement, nonce []byte, intelCache string, referenceValues []ReferenceValue) (*TdxMeasurementResult, bool) {
+	var err error
+	result := &TdxMeasurementResult{}
+	ok := true
+
+	// If the attestationreport does contain neither TDX measurements, nor TDX Reference Values
+	// there is nothing to to
+	if tdxM == nil && len(referenceValues) == 0 {
+		return nil, true
+	}
+
+	if len(referenceValues) == 0 {
+		msg := "Could not find TDX Reference Value"
+		result.Summary.setFalse(&msg)
+		return result, false
+	} else if len(referenceValues) > 1 {
+		msg := fmt.Sprintf("Report contains %v reference values. Currently, only one TDX Reference Value is supported",
+			len(referenceValues))
+		result.Summary.setFalse(&msg)
+		return result, false
+	}
+	tdxReferenceValue := referenceValues[0]
+
+	if tdxReferenceValue.Type != "TDX Reference Value" {
+		msg := fmt.Sprintf("TDX Reference Value invalid type %v", tdxReferenceValue.Type)
+		result.Summary.setFalse(&msg)
+		return result, false
+	}
+	if tdxReferenceValue.Tdx == nil {
+		msg := "TDX Reference Value does not contain policy"
+		result.Summary.setFalse(&msg)
+		return result, false
+	}
+
+	// TODO: support other report types
```

Comment:
Ist das zeitnah realistisch? Sonst könnten wir evtl. auch einfach nur sagen "currently we only support report type v4"